pkg/asset/installconfig/aws/metadata.go

```diff
@@ -139,3 +143,25 @@ func (m *Metadata) VPC(ctx context.Context) (string, error) {
 
 	return m.vpc, nil
 }
+
+// IgnitionPresignedURL returns an S3 bucket name and a presigned URL for it
+func (m *Metadata) IgnitionPresignedURL(ctx context.Context, infraID string) (string, string, error) {
```

Comment:
I don't think this needs to be a member of Metadata. the metadata has GetSession function that you can use to generate the presigned URL.

Reply:
I've moved this to it's own file in the installconfig asset.